packages/hoppscotch-common/src/newstore/settings.ts

```diff
@@ -146,6 +140,26 @@ const dispatchers = defineDispatchers({
 
     return result
   },
+  toggleNestedSetting(
+    currentState: SettingsDef,
+    {
+      settingKey,
+      property,
+    }: { settingKey: KeysMatching<SettingsDef, object>; property: string }
+  ) {
+    const result: Partial<SettingsDef> = {}
+
+    if (!has(currentState, settingKey + "." + property)) {
```

Comment:
```suggestion
    if (!has(currentState, [settingKey, property])) {
```

Let's specify the path as an array.